### PR TITLE
fix(semantic): uniform host gap-fill on blockquote prose lines

### PR DESCRIPTION
**Pull request by atusy**
## Problem

In a blockquote paragraph, identical lines rendered differently: interior `> [foo] aaa` lines lost the `markup.quote` host highlighting over `[` and `] aaa`, while the paragraph's **last** line kept it (screenshot bug report, reproduced from an LSP session log).

## Root cause

The markdown_inline injection registers one exclusion gap per paragraph line. Interior gaps carried their trailing newline, so each spanned `(L,c)..(L+1,0)` — which `filter_by_injection_regions` classifies as a **multiline container** region (host token stripped from content). The last gap ends flush with the text, classifies **single-line**, and keeps the host token as a gap filler.

## Fix

At the exclusion-range push in `rebuild_context`: when the injection's last included range is not newline-terminated (prose-style content, e.g. markdown_inline), trim the trailing newline (LF/CRLF) from every gap so each classifies as the single display line it covers and gap-fills uniformly. Container content (code-fence gaps are newline-terminated on every chunk) is untouched, preserving the "host must not leak into code" design and its tests.

Known accepted edge (documented at the decision site + pinned by a unit test): a fence left unterminated at EOF in a file without a trailing newline ends flush and classifies prose until it closes — at EOF the two shapes are indistinguishable from the ranges alone, and resolving toward container would reinstate the per-line asymmetry for a paragraph ending the file without a trailing newline, the worse artifact.

## Commits

- fix + wire-level e2e regression test reproducing the session log
- fail-soft slicing (`text.get`) matching the function's documented degradation posture
- doc alignment (`filter_by_injection_regions` docs, overlap-resolution ADR, combined-path exemption note)
- unit tests pinning the trim edges (LF/CRLF prose, closed-fence container untrimmed, EOF-flush fence flip)
- EOF-ambiguity rationale comment (codex round)

## Verification

- New e2e test fails on main with the exact reported signatures, passes here
- Full gate: fmt, clippy --all-targets -D warnings, 2776 lib tests, 42 blockquote e2e, full e2e suite 1675 tests green
- Review pipeline: 6-perspective local review (all findings fixed or deliberately kept with rationale), codex converged ("no comments to provide" from a fresh session)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved semantic-token highlighting consistency for blockquote shortcut links across multiple lines.
  * Refined partial injection-region overlap handling with correct prose-gap newline trimming for both LF and CRLF.
  * Preserved newline behavior inside fenced code blocks and correctly trimmed prose gaps for unterminated fences at EOF.
* **Tests**
  * Added end-to-end assertions for uniform blockquote shortcut link highlighting.
  * Added coverage for prose-gap trimming and multiline partial-overlap edge cases.
* **Documentation**
  * Updated partial-overlap decision guidance, including the multiline partial-overlap scenario.

### CODE REVIEW

File: docs/architecture-decisions/semantic-token-overlap-resolution.md [hunk 1]
```diff
@@ -100,11 +100,11 @@ Host tokens (depth=0) overlapping **active** injection regions are handled with
...
 
 The single-line/multiline distinction for partial overlaps identifies the injection's nature:
-- **Single-line regions** arise from direct child injections (e.g., `markdown_inline` within a heading). The host token should fill gaps where the injection produces no tokens (e.g., plain text in headings).
-- **Multiline regions** arise from container injections (e.g., blockquote content). The host token should not leak into the container's content.
+- **Single-line regions** arise from direct child injections (e.g., `markdown_inline` within a heading). The host token should fill gaps where the injection produces no tokens (e.g., plain text in headings). Blockquote *prose* gaps are trimmed of their trailing newline at the exclusion-range push site (`rebuild_context`) precisely so they classify here and gap-fill uniformly on every paragraph line.
+- **Multiline regions** arise from container injections (e.g., blockquoted code fence content). The host token should not leak into the container's content.
```
[the remark this answers is not in the document]
Comment (atusy) [reply docs/architecture-decisions/semantic-token-overlap-resolution.md]: Fixed in 7905360d2 — reworded per the suggestion so the space lives outside the code span.

File: tests/e2e_semantic_blockquote.rs [hunk 2]
```diff
@@ -817,6 +817,124 @@ fn test_blockquote_injection_tokens() {
...
+            tokens.iter().any(|t| t.line == line
+                && t.token_type == string_type
+                && t.start <= 6
+                && t.start + t.length >= 11),
+            "Line {} should keep `string` (markup.quote) coverage over `] aaa`. Tokens: {:?}",
+            line,
+            signature(line)
+        );
+    }
+
```
[the remark this answers is not in the document]
Comment (atusy) [reply tests/e2e_semantic_blockquote.rs:923]: Fixed in ba567f9c8 — the test now also asserts a keyword (markup.link.label) token covering the link label on lines 3/4/6, so a host-only markup.quote result can no longer pass. Full e2e_semantic_blockquote suite green (42).

File: src/analysis/semantic/parallel.rs [hunk 3]
```diff
@@ -850,8 +850,45 @@ fn rebuild_context<'a>(
...
+        // the file without a trailing newline — the worse artifact, since the
+        // fence case renders uniformly and self-heals on close.
+        // `get` instead of indexing: like the content re-slice above, a
+        // corrupt reuse-path entry must degrade (here: to the untrimmed
+        // container classification), not panic.
+        let ends_with_newline = |end: usize| text.get(..end).is_some_and(|s| s.ends_with('\n'));
+        let prose = ranges.last().is_some_and(|r| {
+            text.get(..inj_start_byte + r.end_byte)
+                .is_some_and(|s| !s.ends_with('\n'))
+        });
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/analysis/semantic/parallel.rs:883]: Fixed in e97677780 — both additions use checked_add; a corrupt offset now degrades (range skipped / untrimmed classification) instead of wrapping or panicking.
